### ISSUE TITLE
Vue 3 support

### ISSUE THREAD

**Issue by veritymedia** [yoychen/v-craft#31]
First, I want to say a big thank you for this library, its really awesome!

As the vue world has moved to v3, would it be possible to upgrade this library? Ive poked around, and tried to make it work, but im struggling :/

The main incompatibility is with the new slots system, and v-craft relies on slots heavily.

If someone can point me in the right direction, I could try to figure it out :)

**Comment by rumbis** [comment]
I would also need  someone to point me the right direction for this project with vue3. Thanks in advance

**Comment by loming** [comment]
> I would also need someone to point me the right direction for this project with vue3. Thanks in advance

I'm trying to make it works here https://github.com/loming/v-craft/tree/vue3, it's not fully working yet, but at least the text and properties are showing in the Example now

**Comment by loming** [comment]
It works now in Vue3 :D

> > I would also need someone to point me the right direction for this project with vue3. Thanks in advance
> 
> I'm trying to make it works here https://github.com/loming/v-craft/tree/vue3, it's not fully working yet, but at least the text and properties are showing in the Example now

**Comment by loming** [comment]
@yoychen Do you have plan to supports Vue3 with your great library? Or could I fork it out as a separate project?

**Comment by yoychen** [comment]
@loming 
Of course, the current license of v-craft is MIT, so you can folk it as your need.

We are planning to recreate v-craft in Vue3 and make it more out-of-the-box and progressive. We wish the new v-craft will be a progressive page builder framework. It is not only a toolset for building your custom page builders. It will also provide an out-of-the-box page builder component with some default block elements like [the v-craft sample](https://yoychen.github.io/v-craft/), an interface for encapsulating your custom Vue components, and a plug system library for the ecosystem to create useful plugs.

**Comment by loming** [comment]
great... I'm rewriting your current library in Typescript at the moment, I'll share it to u when it's ready and see if it helps you or not.

I don't have much time to help on redesigning the architecture though... i'm looking forward to your amazing work!

**Comment by genu** [comment]
@loming 
I've been testing on your vue3 branch with nuxt3 and typescript, and I think the naming of the Node class `Node`  causes some issue during compilation.

https://github.com/loming/v-craft/blob/5fd3fccfca90fe0252721bffdcd33be18ada4c6c/src/core/Node.js#L1-L15

 It might be related to SSR. The error is:
![CleanShot 2022-05-17 at 23 40 18@2x](https://user-images.githubusercontent.com/928780/168952588-c840e1bf-cb8b-43e3-b92c-e57252911dfa.png)



I haven't been able to trace down the root cause, but if I rename `Node` to something else, it works.

**Comment by rodenp** [comment]
> @loming Of course, the current license of v-craft is MIT, so you can folk it as your need.
> 
> We are planning to recreate v-craft in Vue3 and make it more out-of-the-box and progressive. We wish the new v-craft will be a progressive page builder framework. It is not only a toolset for building your custom page builders. It will also provide an out-of-the-box page builder component with some default block elements like [the v-craft sample](https://yoychen.github.io/v-craft/), an interface for encapsulating your custom Vue components, and a plug system library for the ecosystem to create useful plugs.

**Comment by rodenp** [comment]
@loming What a fantastic project you have here.
I am using vue3 and am very curious about how you are going with recreating v-craft in Vue3.
I have compared it to the react project -> https://craft.js.org/examples/landing/.
The react project is very well designed and I like the ability to edit the page and then be able to display it when finished.
I also like the toolbars above the content being edited.
It would be great if that could also be in the vue3 project.
It would also be awesome if you could also have layers.
And it would it be great if we were able to export the page as proper html and run it in the browser.
Do you have an update on when you expect to have this ready?
I am very keen to help test it out.

**Comment by rodenp** [comment]
@loming I just tested the vue3 branch
I am on the latest npm and node versions and updated all the npm modules.
It now works a treat
Looking forward to seeing this develop.

**Comment by rodenp** [comment]
What a fantastic project you have here.
I like your vision for the future.
I have compared your project to the react project -> https://craft.js.org/examples/landing/.
The react project is very well designed and I like the ability to edit the page and then be able to display it when finished.
I also like the toolbars above the content being edited.
It would be great if that could also be in the vue3 project.
It would also be awesome if you could also have layers.
And it would it be great if we were able to export the page as proper html and run it in the browser.
Do you have an update on where you are with this project?
I am very keen to help test it out.
And once again keep up the amazing work.

**Comment by khusseini** [comment]
@yoychen, @loming 
I have created a new repository based on your code for Vue3. My version has rewritten most of the code and includes a pinia store for easier statemanagement and integration.

The prototype is working, so far and some features/packages for GraphQL Integration will be also released soon. 
Have a look https://faasaf.gitlab.io/v-craft/main/
https://gitlab.com/faasaf/v-craft

**Comment by yoychen** [comment]
> @yoychen, @loming 
> I have created a new repository based on your code for Vue3. My version has rewritten most of the code and includes a pinia store for easier statemanagement and integration.
> 
> The prototype is working, so far and some features/packages for GraphQL Integration will be also released soon. 
> Have a look https://faasaf.gitlab.io/v-craft/main/
> https://gitlab.com/faasaf/v-craft

Wow, it's awesome. Thanks for sharing 👍💪.

**Comment by DevPlus31** [comment]
Any news ?

**Comment by fusigabs** [comment]
> [@yoychen](https://github.com/yoychen), [@loming](https://github.com/loming) I have created a new repository based on your code for Vue3. My version has rewritten most of the code and includes a pinia store for easier statemanagement and integration.
> 
> The prototype is working, so far and some features/packages for GraphQL Integration will be also released soon. Have a look https://faasaf.gitlab.io/v-craft/main/ https://gitlab.com/faasaf/v-craft

@khusseini - Is this still available? Would like to see your implementation of Pinia store.

**Comment by khusseini** [comment]
> > [@yoychen](https://github.com/yoychen), [@loming](https://github.com/loming) I have created a new repository based on your code for Vue3. My version has rewritten most of the code and includes a pinia store for easier statemanagement and integration.
> > 
> > The prototype is working, so far and some features/packages for GraphQL Integration will be also released soon. Have a look https://faasaf.gitlab.io/v-craft/main/ https://gitlab.com/faasaf/v-craft
> 
> @khusseini - Is this still available? Would like to see your implementation of Pinia store.

My bad, I have moved the repository and reworked the code https://github.com/versa-stack/v-craft as it is part of a "server less" environment I am currently working on. 
The GeaphQl integration has been extracted to its own repository, however I have not published it yet as it still has a lot of issues on the editor side.

**Comment by loming** [comment]
I have made components for the Quasar framework and built some items working with my server-side ETH contract before, but they never reached a point which was good enough to share.

Looking forward to what you build :D, i could transfer the [@v-craft3/core](https://www.npmjs.com/package/@v-craft3/core) to  you as well if you want

**Comment by khusseini** [comment]
@loming thank you for your interest and offer to transfer your repository to me, however the two codebases differ a lot as I have rewritten the core from the ground up. I currently working on chores to make github actions for tests, releases and docs building. Contributors are always welcome :)

**Comment by fusigabs** [comment]
BTW @khusseini were you able to implement switching designs between devices in your rewrite (ie specifying styles for Desktop, Tablet and Mobile)?

**Comment by khusseini** [comment]
@nfunwigabga I'd expect that to be handled by the implementation of the editor, not by the editor itself, however feel free to submit an issue/feature request on my repository. How about we continue the conversation over there https://github.com/versa-stack/v-craft/discussions? I can also try to guide you through an implementation

**Comment by Milanator** [comment]
Is there any updates?
